lib/src/server/interceptor.dart

```diff
@@ -0,0 +1,13 @@
+import '../shared/status.dart';
+
+import 'call.dart';
+import 'service.dart';
+
+/// A gRPC Interceptor.
+///
+/// An interceptor is called before the corresponding [ServiceMethod] invocation.
+/// If the interceptor returns a [GrpcError], the error will be returned as a response and [ServiceMethod] wouldn't be called.
+/// If the interceptor throws [Exception], [GrpcError.internal] with exception.toString() will be returned.
+/// If the interceptor returns null, the corresponding [ServiceMethod] of [Service] will be called.
+typedef Interceptor = GrpcError Function(
```

Comment:
@sigurdm I have checked go implementation for interceptors

https://github.com/grpc/grpc-go/blob/a3e99ebee0a498c3093e9741d09c579f0fc34a7f/server.go#L1020

From my understanding it has only two cases - `no-error` and `error`, 
if there is an error, write a status and stop processing a request.
If there is no error, continue processing.

Imo it doesn't make sense to add this feature (I mean silent blocking) now.
Also I am not sure it conforms to grpc standard or not. 

What do you think? Is it required?

Reply:
No let's stick with error/no-error!